packages/mui-material/src/Grid/Grid.d.ts

```diff
@@ -10,7 +17,71 @@ export type GridSpacing = number | string;
 
 export type GridWrap = 'nowrap' | 'wrap' | 'wrap-reverse';
 
-export type GridSize = 'auto' | number;
+export type GridSize = 'auto' | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
```

Comment:
```suggestion
export type GridSize = 'auto' | number;
```

We shouldn't change this, the code diff in the comment of the issue is outdated.